### ISSUE TITLE
chdir fails with 210/CHROOT (instead of 200/CHDIR)

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#5125]
Inspired by https://github.com/systemd/systemd/issues/5124

```sh
[Service]
Type=oneshot

ExecStart=/bin/bash -c 'echo hello'
WorkingDirectory=~
```

```sh
test.service: Trying to enqueue job test.service/start/replace
test.service: Installed new job test.service/start as 58
test.service: Enqueued job test.service/start as 58
test.service: Passing 0 fds to service
test.service: About to execute: /bin/bash -c 'echo hello'
test.service: Forked /bin/bash as 200
test.service: Changed dead -> start
Starting test.service...
test.service: Child 200 belongs to test.service
test.service: Main process exited, code=exited, status=210/CHROOT
test.service: Changed start -> failed
test.service: Job test.service/start finished, result=failed
Failed to start test.service.
test.service: Unit entered failed state.
test.service: Failed with result 'exit-code'.
test.service: cgroup is empty
```

This regression introduced in e7f1e7c6e2f91f3cad5e

**Comment by fsateler** [comment]
Introduced in e7f1e7c6e2f91f3cad5eadfcc6ab9673caedb838

cc @tixxdz

**Comment by keszybz** [comment]
Any progress?

**Comment by poettering** [comment]
Fix pending in #5287.
